pkg/things/coapclient.go

```diff
@@ -195,7 +195,15 @@ func (c *IECClient) AccessToken(_ string, jws string) (reply []byte, err error)
 	ctx, cancel := c.context()
 	defer cancel()
 
-	response, err := conn.PostWithContext(ctx, "/accesstoken", coap.AppJSON, strings.NewReader(jws))
+	var coapFormat coap.MediaType
+	switch format {
+	case commandFormatJOSE:
+		coapFormat = coap.AppCoseSign
```

Comment:
The CoAP Content-Formats registry does not contain a value for JOSE content so for the interim I have used `cose-sign` to indicate a signed JWT.

Reply:
Should we pick an unassigned number from that registry (maybe in the 11544-64999 range) and assign it to `application/jose`? For example: 
```
const AppJOSE coap.MediaType = 11650
```
That way we leave `cose-sign` open to use if/when we add cose support for constrained devices.

Reply:
I'll add new `const AppJoseSign coap.MediaType` to be consistent with the other media types and to give us the option to differentiate between signing and encryption.